src/Core/Util/InternationalizedDomainNameConverter.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Util;
+
+final class InternationalizedDomainNameConverter
+{
+    /**
+     * Convert the host part of the email from punycode to utf8 (e.g,. email@xn--e1aybc.xn--p1ai -> email@тест.рф)
+     *
+     * @param string $email
+     *
+     * @return string
+     */
+    public function emailToUtf8(string $email): string
+    {
+        $parts = explode('@', $email);
+        if (count($parts) !== 2) {
+            return $email;
+        }
+
+        return $parts[0] . '@' . idn_to_utf8($parts[1], 0, INTL_IDNA_VARIANT_UTS46);
```

Comment:
Watch out with `INTL_IDNA_VARIANT_UTS46` it's only available as of ICU 4.6
Because we now use SwiftMailer > 6.2 we should have a look on what SwiftMailer can do 
https://github.com/swiftmailer/swiftmailer/pull/1044

Reply:
@PierreRambaud I think we're pretty safe because we have the [Symfony polyfill](https://github.com/symfony/polyfill-intl-idn). WDYT?

Reply:
You're right :+1: